org-code-javabuilder/README.md

```diff
@@ -1,33 +1,138 @@
-This will exercise the compile and run logic in isolation. Javabuilder will execute the
-program in src/main/resources/main.json. You can update which file is used from `LocalProjectFileManager`.
-Input and output will be directed to the terminal.
-Any non-Java file used or created by the program (for example `grid.txt` in `main_painter.json`) will be
-created in the `org-code-javabuilder` folder when running locally, so be aware you may want 
-to do some cleanup after running code that relies on text files (in our Lambda runtime the code 
-all runs from a temporary folder and handles this cleanup). 
-
-### Developing with Dashboard
+# Javabuilder build-and-run package
+This portion of Javabuilder contains the code that compiles and runs student code as well
+as the code for the `org.code` Java packages that are used in the Java Lab mini apps.
+
+### Directory
+* [lib](https://github.com/code-dot-org/javabuilder/tree/main/org-code-javabuilder/lib)
+  contains the "backend" code that compiles, builds, and runs student code. This is not
+  user-facing. It also handles the WebSocket layer that communicates between the client
+  and the running program. Also contained here are implementations of the communication
+  protocol and program initialization that are specific to local development and AWS
+  development. Local implementations are in the `dev.javabuilder` package while AWS
+  implementations are in the main `org.code.javabuilder` package and are prefixed with
+  `AWS` or `Lambda`
+* [media](https://github.com/code-dot-org/javabuilder/tree/main/org-code-javabuilder/media)
+  contains a package that can be imported by the user to edit sounds, colors, fonts, and
+  images.
+* [neighborhood](https://github.com/code-dot-org/javabuilder/tree/main/org-code-javabuilder/neighborhood)
+  contains a package that can be imported by the user to interact with the neighborhood
+  mini-app, a maze-like app created for AP CSA.
+* [protocol](https://github.com/code-dot-org/javabuilder/tree/main/org-code-javabuilder/protocol)
+  contains the communication protocol that handles messages between Java Lab (client) and
+  Javabuilder (server)
+* [theater](https://github.com/code-dot-org/javabuilder/tree/main/org-code-javabuilder/theater)
+  contains a package that can be imported by the user to interact with the theater 
+  mini-app, an image and sound editing app created for AP CSA and used in standalone Java
+  Lab projects.
+
+## Local Development
+### Setup your project
+1. Install a Java IDE - We recommend IntelliJ Community edition.
+    1. If given the option when installing, select Java 11 as the default
+1. If Java 11 was not installed as part of the IDE, install 
+   [OpenJDK 11](https://jdk.java.net/java-se-ri/11)
+1. Download the repo and build. Building is a one-time step that seeds the git commit
+   hooks.
+```
+git clone git@github.com:code-dot-org/javabuilder.git
+cd javabuilder/org-code-javabuilder
+./gradlew build
```

Comment:
on my computer (mac) I can also run `gradle build` (similar for  appRun, etc.)

Reply:
nit: also might be worth mentioning that you can run gradle commands inside IntelliJ since we are recommending IntelliJ as the IDE to use

Reply:
How do you run gradle commands inside IntelliJ? I always just do it from the command line.

Reply:
There is a "Gradle" tool that can be accessed via the sidebar (I think it may have to be enabled), but from there you can click the elephant gradle icon to run anything, after which recent invocations will appear in your Select Run/Debug Configuration Menu

![Screen Shot 2021-09-17 at 6 36 51 PM](https://user-images.githubusercontent.com/85528507/133867922-aebd6405-7058-4898-8191-a54f2c0f7580.png)
![Screen Shot 2021-09-17 at 6 37 05 PM](https://user-images.githubusercontent.com/85528507/133867925-b08bd84e-4c22-458d-a81f-f8a4548616c1.png)
![Screen Shot 2021-09-17 at 6 37 21 PM](https://user-images.githubusercontent.com/85528507/133867939-8fccc65a-350a-4d7a-8a81-966e0e0bc50f.png)
![Screen Shot 2021-09-17 at 6 37 30 PM](https://user-images.githubusercontent.com/85528507/133867941-76e887d7-5d18-4c16-9982-d5183752e20b.png)
![Screen Shot 2021-09-17 at 6 37 52 PM](https://user-images.githubusercontent.com/85528507/133867947-b16163bd-8355-4ceb-a52a-e376173a2090.png)